app/services/hotkeys.ts

```diff
@@ -40,148 +53,177 @@ export interface IBinding {
   };
 }
 
-
 interface IHotkeyAction {
   name: string;
+
   description(entityId: string): string;
-  down(entityId: string): void;
+
+  down(entityId: string, hotkeyId?: number): void;
+
   isActive?(entityId: string): boolean;
+
   shouldApply?(entityId: string): boolean;
-  up?(entityId: string): void;
+
+  up?(entityId: string, hotkeyId?: number): void;
 
   // These are injected dynamically
   downHandler?(): void;
+
   upHandler?(): void;
 }
 
+type HotkeyGroup = {
+  [x: string]: IHotkeyAction;
+};
 
-/**
- * All possible hotkeys should be defined in this object.
- * All information about the hotkey and its behavior is
- * encapsulated within the action definition here.
- *
- * WARNING: Changing the name of existing hotkey actions
- * will cause people to lose their saved keybindings. The
-* name shouldn't really change after it is added.
-*/
-const HOTKEY_ACTIONS: Dictionary<IHotkeyAction[]> = {
-  GENERAL: [
-    {
-      name: 'TOGGLE_START_STREAMING',
-      description: () => $t('Start Streaming'),
-      down: () => getStreamingService().toggleStreaming(),
-      isActive: () => {
-        const streamingService = getStreamingService();
-        return streamingService.isStreaming;
-      }
-    },
-    {
-      name: 'TOGGLE_STOP_STREAMING',
-      description: () => $t('Stop Streaming'),
-      down: () => {
-        const streamingService = getStreamingService();
-        streamingService.toggleStreaming();
-      },
-      isActive: () => {
-        const streamingService = getStreamingService();
-        return !streamingService.isStreaming;
-      }
-    },
-    {
-      name: 'TOGGLE_START_RECORDING',
-      description: () => $t('Start Recording'),
-      down: () => getStreamingService().toggleRecording(),
-      isActive: () => getStreamingService().isRecording
+const GENERAL_ACTIONS: HotkeyGroup = {
+  TOGGLE_START_STREAMING: {
+    name: 'TOGGLE_START_STREAMING',
+    description: () => $t('Start Streaming'),
+    down: () => getStreamingService().toggleStreaming(),
+    isActive: () => {
+      const streamingService = getStreamingService();
+      return streamingService.isStreaming;
     },
-    {
-      name: 'TOGGLE_STOP_RECORDING',
-      description: () => $t('Stop Recording'),
-      down: () => getStreamingService().toggleRecording(),
-      isActive: () => !getStreamingService().isRecording
+  },
+  TOGGLE_STOP_STREAMING: {
+    name: 'TOGGLE_STOP_STREAMING',
+    description: () => $t('Stop Streaming'),
+    down: () => {
+      const streamingService = getStreamingService();
+      streamingService.toggleStreaming();
     },
-    {
-      name: 'ENABLE_STUDIO_MODE',
-      description: () => $t('Enable Studio Mode'),
-      down: () => getTransitionsService().enableStudioMode(),
-      isActive: () => getTransitionsService().state.studioMode
+    isActive: () => {
+      const streamingService = getStreamingService();
+      return !streamingService.isStreaming;
```

Comment:
Don't see a reason why you replaced one line expression to the 2 lines

Reply:
Can you please paste the line? I'm not entirely sure what you're referring to in this particular case.